frontend/hooks/useBuildLogs.js

```diff
@@ -35,8 +39,14 @@ export const useBuildLogs = (id) => {
         }
       }
 
-      if (outputCount > 0 && ['success', 'error'].includes(state)) {
-        setResults(updatedResults);
+      if (['success', 'error'].includes(state)) {
+        if (output?.length > 0) {
+          setResults(updatedResults);
+        }
+
+        if (output?.length === 0 && results.totalRequests === 0) {
```

Comment:
Should this be `results.totalRequests > 0`? My assumption is that we want to still set the results if we didn't get output but have made at least one request

Reply:
There are situations where errored builds may or may not have logs so we update the state of just the initial request if there are no logs. If it was `results.totalRequests > 0` it would cause an infinite loop since we fetch the api results recursively.